### PR TITLE
Multiformat ad unit

### LINKED ISSUE #688: Multiformat ad unit for Original API
## Objectives

Please use this ticket in the iOS repository as a source of requirements: https://github.com/prebid/prebid-mobile-ios/issues/913

If any Android-specific issues arise let's discuss them here.

### PR DESCRIPTION
**Pull request by ValentinPostindustria**
Closes #688.

### CODE REVIEW

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/api/original/PrebidAdUnit.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    ) {
+        baseFetchDemand(request, null, listener);
+    }
+
+    public void fetchDemand(
+            PrebidRequest request,
+            Object adObject,
+            OnFetchDemandResult listener
+    ) {
+        baseFetchDemand(request, adObject, listener);
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/api/original/PrebidAdUnit.java:37]: Need to change the order of parameters

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java [hunk 2]
```diff
@@ -109,7 +124,7 @@ public void fetchDemand(@NonNull final OnCompleteListener2 listener) {
         });
     }
 
-    public void fetchDemand(@NonNull Object adObj, @NonNull OnCompleteListener listener) {
+    public void fetchDemand(@Nullable Object adObj, @NonNull OnCompleteListener listener) {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java:127]: Why do we change the notation to Nullable? What is the use case? And, can we avoid it? 

It's a public API, and for the publishers, this method should always _demand_ an ad object. 

I see that it's due to migration to BidInfo. However, let's think about how we can avoid changing the public API.

**Review by YuriyVelichkoPI — CHANGES_REQUESTED** [review]

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/api/data/BidInfo.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.annotation.Nullable;
+
+import org.prebid.mobile.ResultCode;
+
+import java.util.Map;
+
+public class BidInfo {
+
+    @NonNull
+    private ResultCode resultCode;
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/api/data/BidInfo.java:13]: Please, rename to `result`

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java (hunk 2 above)
Comment (ValentinPostindustria) [reply PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java]: We use this method in the new multiformat API so it's possible that it's null. I removed this annotation and now old users will not think that nullable is okay.

**Pull request merged**